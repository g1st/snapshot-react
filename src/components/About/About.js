import React from 'react';

import img from '../../assets/images/about-woman-img.jpg';
import {
  Wrapper,
  AboutWrap,
  Background,
  HeadingText,
  Text,
  CircleWrapper,
  Circle
} from './styled';

const About = () => {
  return (
    <Wrapper>
      <AboutWrap>
        <Background>
          <HeadingText>Background</HeadingText>
          <Text>
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Iste
            minima expedita aut voluptatem praesentium quae, ad veniam eveniet
            eum quaerat. Lorem ipsum dolor sit amet consectetur adipisicing
            elit.
          </Text>
        </Background>
        <CircleWrapper>
          <Circle src={img} alt="Woman in dress" />
        </CircleWrapper>
      </AboutWrap>
    </Wrapper>
  );
};

export default About;
