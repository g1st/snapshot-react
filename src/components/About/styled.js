import styled from 'styled-components';
import { darken } from 'polished';

import img from '../../assets/images/about-woman-img.jpg';

import {
  colorWhiteBg,
  colorText,
  colorMainLight
} from '../../styles/variables';

export const Wrapper = styled.div`
  background-color: ${colorWhiteBg};
  padding: 80px 0;
`;
Wrapper.displayName = 'Wrapper';

export const Background = styled.div`
  background-color: #fff;
  margin: 0 15px;
  border-radius: 5px;
  padding: 30px;
`;
Background.displayName = 'Background';

export const HeadingText = styled.h1`
  text-transform: uppercase;
  text-align: center;
  font-weight: 300;
  letter-spacing: 6px;
  font-size: 1.8rem;
  line-height: 86px;
  padding-bottom: 20px;
  color: ${colorMainLight};
  @media (min-width: 500px) {
    letter-spacing: 14px;
    font-size: 2.4rem;
  }
`;
HeadingText.displayName = 'HeadingText';

export const Text = styled.p`
  color: ${darken(0.2, `${colorText}`)};
  font-size: 1.2rem;
  text-align: center;
  letter-spacing: 2px;
  line-height: 32px;
  padding-bottom: 20px;
`;
Text.displayName = 'Text';

export const CircleWrapper = styled.div`
  padding-right: 15px;
  padding-left: 15px;
`;
CircleWrapper.displayName = 'CircleWrapper';

export const Circle = styled.img`
  height: auto;
  max-width: 100%;
  padding: 15px;
  background-position: center;
  background-repeat: no-repeat;
  border-radius: 50%;
  display: block;
  margin: 50px auto 0 auto;
  border: 10px solid #fff;
`;
Circle.displayName = 'Cricle';
