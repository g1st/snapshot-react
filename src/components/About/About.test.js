import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import About from '.';

describe('About Component', () => {
  it('matches snapshot', () => {
    const tree = shallow(<About />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it('Wrapper must have #about  id attribute', () => {
    const tree = shallow(<About />);
    expect(tree.find('Wrapper').find('#about')).toHaveLength(1);
  });
});
