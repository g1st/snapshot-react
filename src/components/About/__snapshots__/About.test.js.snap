// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Component matches snapshot 1`] = `
<Wrapper
  id="about"
>
  <AboutWrap>
    <Background>
      <HeadingText>
        Background
      </HeadingText>
      <Text>
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Iste minima expedita aut voluptatem praesentium quae, ad veniam eveniet eum quaerat. Lorem ipsum dolor sit amet consectetur adipisicing elit.
      </Text>
    </Background>
    <CircleWrapper>
      <Cricle
        alt="Woman in dress"
        src="about-woman-img.jpg"
      />
    </CircleWrapper>
  </AboutWrap>
</Wrapper>
`;
